packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { isNullOrUndefined } from 'util';
+import * as vscode from 'vscode';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '../commands';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+
+export class ForceListMetadataExecutor extends SfdxCommandletExecutor<string> {
+  private metadataType: string;
+  private outputPath: string;
+  private defaultUsernameOrAlias: string;
+
+  public constructor(
+    metadataType: string,
+    outputPath: string,
+    defaultUsernameOrAlias: string
+  ) {
+    super();
+    this.metadataType = metadataType;
+    this.outputPath = outputPath;
+    this.defaultUsernameOrAlias = defaultUsernameOrAlias;
+  }
+
+  public build(data: {}): Command {
+    let builder = new SfdxCommandBuilder()
+      .withArg('force:mdapi:listmetadata')
```

Comment:
how about having this as a const?

Reply:
I didn't save it as a constant to follow the format of the other commands in our core package. However, I realized that the work item for pr specifies that we do want to display the progress notification for the command to the user. In that case, should we also specify a description field for the command builder?